### PR TITLE
`save` command: Don't use BufWriter to write external strem to a file

### PR DESCRIPTION
**Pull request by WindSoilder**
# Description
Fixes: #9293

The problem is caused by `save` makes a `BufferWriter` for output file, when external commands redirect it's output to a file, the content is bufferred first...

To fix the issue, I'd like to introduce a `--no-buf` flag for `save` command, and it's only used in redirection scenario.

Sorry it's hard to test against it in test, because it requires external command to sleep or pause... 

# User-Facing Changes

# Tests + Formatting

# After Submitting

### CODE REVIEW

**Review by amtoine — COMMENTED** [review]
i've been able to test the use case of https://github.com/nushell/nushell/issues/9293 and this very PR fixes the issue :ok_hand: 

my only question is: **do we need to have `--no-buf` if it's only used internally?** we could just have it hardcoded to the its default value, right?

**Comment by WindSoilder** [comment]
> we could just have it hardcoded to the its default value, right?

I have thought about it at first time, but finally I don't think it's a good idea...If user want to run `save` directly like `python a.py | save -r a.txt`.  I'd expect it runs into complete at once, which means I care less about the previous commands running status.  Make it by default would cause more IO, which harms performance..

File: crates/nu-command/src/filesystem/save.rs [hunk 1]
```diff
@@ -349,7 +349,11 @@ fn stream_to_file(
     span: Span,
     progress: bool,
 ) -> Result<PipelineData, ShellError> {
-    let mut writer = BufWriter::new(file);
+    // Note: when saved from external stream to file, no need to use BufWriter
+    // so we can see the output of external stream in a file immediately.
+    //
+    // It's good for users to see running status of external program.
```
Comment (amtoine) [inline crates/nu-command/src/filesystem/save.rs:355]: i'm personnaly not a huge fan of comments in code, they can be hard to maintain and evolve with the code... :confused: 

maybe we can just link to this PR, which already contains this information? :yum: 

in the end, i do not have a super strong opinion on this, as this change is not super trivial :wink:

**Review by amtoine — COMMENTED** [review]
wow that's super simple :open_mouth: 

let me try again real quick :relieved:

File: crates/nu-command/src/filesystem/save.rs (hunk 1 above)
Comment (WindSoilder) [reply crates/nu-command/src/filesystem/save.rs]: Thanks, I think you're right :-D  Adjusted the comment

Comment (amtoine) [reply crates/nu-command/src/filesystem/save.rs]: cool :yum:

**Review by amtoine — APPROVED** [review]
thanks a ton @WindSoilder, i think we can land this :partying_face:
